src/Config/FileSystemRuntimeConfigLoader.cs

```diff
@@ -87,45 +90,76 @@ public string GetConfigFileName()
         return configFileName;
     }
 
+    /// <summary>
+    /// Checks if we have already attempted to configure the file watcher, if not
+    /// instantiate the file watcher if we are in the development mode.
+    /// Returns true if we instantiate a new file watcher.
+    /// </summary>
+    private bool TrySetupConfigFileWatcher()
+    {
+        if (_configFileWatcher is not null)
+        {
+            return false;
+        }
+
+        if (RuntimeConfig is not null && RuntimeConfig.IsDevelopmentMode())
+        {
+            try
+            {
+                _configFileWatcher = new(this, GetConfigDirectoryName(), GetConfigFileName());
```

Comment:
It would be good to have checkpoint logs indicating reload.

Reply:
do you mea log that we are in a hot-reload scenario here because we are instantiating the filewatcher?

Reply:
will add to the hot-reload function